### PR TITLE
utils: make `-BuildTargets` optional in build.ps1

### PR DESCRIPTION
**Pull request by compnerd**
This doesn't really add much to readability and is almost always `default`. Simply assume the default behaviour and allow callers to alter the behaviour if needed.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @hyp @hjyamauchi

File: utils/build.ps1 [hunk 1]
```diff
@@ -1728,7 +1723,6 @@ function Build-Dispatch([Platform]$Platform, $Arch, [switch]$Test = $false) {
     -Arch $Arch `
     -Platform $Platform `
     -UseBuiltCompilers C,CXX,Swift `
```
Comment (hjyamauchi) [inline utils/build.ps1:1725]: Is it okay not to keep `ExperimentalTest` when testing? If `$Target` is really dead, better to remove it from the above statement?

File: utils/build.ps1 [hunk 2]
```diff
@@ -1782,7 +1775,7 @@ function Build-Foundation([Platform]$Platform, $Arch, [switch]$Test = $false) {
         -Arch $Arch `
         -Platform $Platform `
         -UseBuiltCompilers ASM,C,CXX,Swift `
-        -BuildTargets $Targets `
+        -SwiftSDK:$SDKRoot `
```
Comment (hjyamauchi) [inline utils/build.ps1:1778]: Is this `-SwiftSDK` flag fixing something or was it optional?

File: utils/build.ps1 [hunk 3]
```diff
@@ -1882,7 +1871,7 @@ function Build-XCTest([Platform]$Platform, $Arch, [switch]$Test = $false) {
       $env:Path = "$XCTestBinaryCache;$FoundationBinaryCache\bin;$DispatchBinaryCache;$(Get-TargetProjectBinaryCache $Arch Runtime)\bin;$env:Path;$UnixToolsBinDir"
     } else {
       $TestingDefines = @{ ENABLE_TESTING = "NO" }
-      $Targets = @("default")
+      $Targets = @("install")
```
Comment (hjyamauchi) [inline utils/build.ps1:1874]: Could this `$Target` to be empty instead of having "install" like other places?

File: utils/build.ps1 (hunk 2 above)
Comment (compnerd) [reply utils/build.ps1:1793]: It is an optional flag; it seems that we cannot get away with it being set for Windows. I think that making this build always in that mode would be nice, but beyond the scope of the cleanup.

File: utils/build.ps1 (hunk 3 above)
Comment (compnerd) [reply utils/build.ps1:1890]: Yes, this side can be empty, good idea.

File: utils/build.ps1 (hunk 1 above)
Comment (compnerd) [reply utils/build.ps1:1737]: No, that needs to be preserved to run the tests.

Comment (hjyamauchi) [reply utils/build.ps1:1737]: So, we need to keep the `-BuildTargets $Targets` here?

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

**Comment by compnerd** [comment]
@swift-ci please clean test macOS platfoorm

**Comment by compnerd** [comment]
@swift-ci please clean test macOS platform

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

**Pull request merged**